demo-contact-diary-app/src/main/java/iris/demo/contact_diary_app/config/RPCClientConfig.java

```diff
@@ -0,0 +1,81 @@
+package iris.demo.contact_diary_app.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.googlecode.jsonrpc4j.JsonRpcHttpClient;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.ConstructorBinding;
+import org.springframework.context.annotation.Bean;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+
+@ConstructorBinding
+@RequiredArgsConstructor
+@ConfigurationProperties("eps-client")
+@Getter
+public class RPCClientConfig {
+
+	private final @NonNull String proxyClientUrl;
+
+	@Bean
+	public JsonRpcHttpClient proxyRpcClient()
+			throws NoSuchAlgorithmException, KeyManagementException, MalformedURLException {
+		return rpcClient(proxyClientUrl);
+	}
+
+	public JsonRpcHttpClient rpcClient(String clientUrl) throws MalformedURLException, NoSuchAlgorithmException, KeyManagementException {
```

Comment:
The configured bean is not used, is it?

You can declare the second method `public JsonRpcHttpClient rpcClient(String clientUrl)` with:
```
@Bean
@Scope(BeanDefinition.SCOPE_PROTOTYPE)
```
and use this with inject:
```
ObjectProvider<JsonRpcHttpClient> rpcClientProvider;
```
and call:
```
var rcpClient = rpcClientProvider.getObject(url);
```

Reply:
Done.